### PR TITLE
i18n support and tooling

### PR DESCRIPTION
**Pull request by OmarIthawi**
I'd like to translate this XBlock and add Japanese as a supported language.

Before you continue, take a look at [edx/i18n-tools](https://github.com/edx/i18n-tools) and [edx/cookiecutter-django-app](https://github.com/edx/cookiecutter-django-app).

### Summary of the Changes
 - Added a Makefile to help manage the translations. The file is copied and modified from: [edx/cookiecutter-django-app](https://github.com/edx/cookiecutter-django-app)
 - Created a `locale` link to `pdf/translations` to allow the `i18n_tools` find the po/mo files
 - Added `pdf/translations/config.yaml` which works as the configuration file of the `i18n_tools` command 
 - Added `requirements-dev.txt` to pin development packages. Used for now only for translations.

### TODO

 - [x] Translate all strings in HTML
 - [x] Create a project in Transifex (or POEditor.com)
 - [x] Document how to update the strings properly from Transifex
 - [x] Manually test that the dummy Esperanto translation is working
 - [ ] @OmarIthawi to Take a look at: https://github.com/IONISx/xblock-pdf

### CODE REVIEW

**Comment by OmarIthawi** [comment]
@melvinsoft Could you please take a look here?

**Comment by MarCnu** [comment]
I'll look at it tomorrow!

Le mar. 11 sept. 2018 21:25, Omar Al-Ithawi <notifications@github.com> a
écrit :

**Comment by OmarIthawi** [comment]
Thanks a lot @MarCnu :slightly_smiling_face: 

Perhaps you've noticed that now this repo has it's own Transifex repo:
 - https://www.transifex.com/open-edx/xblocks/ (you need to be logged in to see the actual resources)

**Review by melvinsoft — COMMENTED** [review]
@OmarIthawi I started testing the Xblock in our Ficus devstack, and I've found a couple of issues with the requirements, after perform the installation following the README, the following requirements were updated and I needed to re install them manually.

`fs==2.1.1` 
and in Ficus we need
`fs==0.4.0`

`lxml==3.4.4`
and in Ficus we need
`lxml==3.4.4`

`python-dateutil==2.7.3`
and in Ficus we need
`python-dateutil==2.1`

Please let me know your thoughts

**Comment by OmarIthawi** [comment]
Thanks @melvinsoft for you review. I've updated the requirements file to match what's in the [Google Drive XBlock](https://github.com/edx-solutions/xblock-google-drive/pull/27/files#diff-d9383e4fc78d9b72886e207884f8f14a).

However the `requirements-dev.txt` shouldn't make it into the Ficus devstack, in production at least that's the case.

File: setup.py [hunk 1]
```diff
@@ -29,5 +29,5 @@ def package_data(pkg, root):
             'pdf = pdf:pdfXBlock',
         ]
     },
-    package_data=package_data("pdf", "static"),
-)
\ No newline at end of file
+    package_data=package_data("pdf", "static", "translations"),
```
Comment (melvinsoft) [inline setup.py]: @OmarIthawi There is a format error here, it should be `package_data=package_data("pdf", ["static", "translations"]),` and also change in the package data loop, to process the roots, like in the oppia XBlock:

```
def package_data(pkg, roots):
     """Generic function to find package_data.
 
     All of the files under each of the `roots` will be declared as package
     data for package `pkg`.
 
     """
     data = []
     for root in roots:
         for dirname, _, files in os.walk(os.path.join(pkg, root)):
             for fname in files:
                 data.append(os.path.relpath(os.path.join(dirname, fname), pkg))
 
     return {pkg: data}
```

**Review by melvinsoft — CHANGES_REQUESTED** [review]
@OmarIthawi I've found a couple of problems in the last changes on the `setup.py` please check my latests inline comments.

**Comment by OmarIthawi** [comment]
Thanks @melvinsoft! I've updated the PR. Please take another look.

**Comment by melvinsoft** [comment]
@OmarIthawi Thanks for addressing the change, the install work good now. But I'm still experiencing error mentioned above. When I install the XBlock on my (Ficus) devstack, is upgrading three dependencies, that start conflicting with the edX one, and I cannot load the LMS anymore:

fs==2.1.1
and in Ficus we need
fs==0.4.0

lxml==3.4.4
and in Ficus we need
lxml==3.4.4

python-dateutil==2.7.3
and in Ficus we need
python-dateutil==2.1

☝️ Those are the upgraded dependencies, this isn't happening when I install the dev requirements, is happening when I install the XBlock. I'm gonna paste the commands to be sure:

```
sudo su edxapp -s /bin/bash
source /edx/app/edxapp/edxapp_env
cd /edx/src
pip install pdfXBlock/ --upgrade
```

**Comment by OmarIthawi** [comment]
Thanks @melvinsoft. Please don't install using `pip install pdfXBlock/ --upgrade`. It should be fine if you refrain from using `--upgrade`.

**Comment by melvinsoft** [comment]
@OmarIthawi 🤦‍♂️ Thanks! I think that was the problem. Sorry my mistake.

**Review by melvinsoft — APPROVED** [review]

**Comment by melvinsoft** [comment]
@OmarIthawi Amazing work here! I'm approving the PR!

**Comment by OmarIthawi** [comment]
Thanks @melvinsoft for the review here!

**Comment by OmarIthawi** [comment]
@MarCnu would you like to get this merged in this repo? I would love to get it merged here so we don't end up in an synchronized forks.

As you can see the translation would end up being here as well, so I'll probably open one or two more pull requests to update the translations.

Depending on how would you like to move forward I see a couple of options:

 - Get it merged here
 - I take a hike and fork it :sweat_smile: 
 - I'm totally fine with getting write access to this repo if you'd like me to

Please let me know what do you think!

**Comment by OmarIthawi** [comment]
Thank you @melvinsoft and @MarCnu. I'm closing this PR in favor of https://github.com/appsembler/pdfXBlock/pull/1.

**Pull request closed without merging**